craft_application/application.py

```diff
@@ -294,6 +295,20 @@ def run(self) -> int:
             return_code = 128 + signal.SIGINT
         except craft_cli.CraftError as err:
             self._emit_error(err)
+        except craft_parts.PartsError as err:
+            self._emit_error(
+                craft_cli.CraftError(
+                    err.brief, details=err.details, resolution=err.resolution
+                )
+            )
+            return_code = 256
+        except craft_providers.ProviderError as err:
+            self._emit_error(
+                craft_cli.CraftError(
+                    err.brief, details=err.details, resolution=err.resolution
+                )
+            )
+            return_code = 512
```

Comment:
```suggestion
        except craft_providers.ProviderError as err:
            self._emit_error(
                craft_cli.CraftError(
                    err.brief, details=err.details, resolution=err.resolution
                )
            )
```

I'm not certain how Python's implementation, Linux's implementation, and architecture endianness might interact with exit codes that don't fit in a byte. On my machine this ends up as 0 though, which isn't desired.

We'd have to do more work to sort out a decent [standard exit code](https://manpages.ubuntu.com/manpages/lunar/en/man3/sysexits.h.3head.html) from parts and provider errors, and that's a separate problem to tackle if we need to.

Reply:
catch